src/PoolManager.sol

```diff
@@ -279,21 +278,24 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
     }
 
     /// @inheritdoc IPoolManager
-    function settle(Currency currency) external payable override onlyWhenUnlocked returns (uint256 paid) {
+    function settle() external payable onlyWhenUnlocked returns (uint256 paid) {
+        Currency currency = Reserves.getSyncedCurrency();
         if (currency.isNative()) {
             paid = msg.value;
         } else {
             if (msg.value > 0) NonZeroNativeValue.selector.revertWith();
-            uint256 reservesBefore = currency.getReserves();
-            uint256 reservesNow = sync(currency);
+            // Reserves are guaranteed to be set, because currency and reserves are always set together
+            uint256 reservesBefore = Reserves.getReserves();
+            uint256 reservesNow = currency.balanceOfSelf();
             paid = reservesNow - reservesBefore;
         }
 
         _accountDelta(currency, paid.toInt128(), msg.sender);
+        Reserves.reset();
```

Comment:
I had a question - if we move this up into the else clause.. Does that basically re introduce the issue?

Reply:
good catch